psi4/driver/procrouting/solvent/pol_embed.py

```diff
@@ -0,0 +1,153 @@
+
+    pol_embed_options.do_diis = core.get_local_option('PE', 'DIIS')
+    pol_embed_options.maxiter = core.get_local_option('PE', 'MAXITER')
+    pol_embed_options.pe_border = core.get_local_option('PE', 'BORDER')
+
+    if pol_embed_options.pe_border:
+        pol_embed_border_options = cppe.PeBorderOptions()
+        pe_btype = core.get_local_option('PE', 'BORDER_TYPE').upper()
+        if pe_btype == "REMOVE":
+            pol_embed_border_options.border_type = core.PeBorderOptions.BorderType.rem
+        elif pe_btype == "REDIST":
+            pol_embed_border_options.border_type = core.PeBorderOptions.BorderType.redist
+        pol_embed_border_options.rmin = core.get_local_option('PE', 'BORDER_RMIN')
+        if core.get_local_option('PE', 'BORDER_RMIN_UNIT').upper() == "AA":
+            pol_embed_border_options.rmin *= 1.0 / constants.bohr2angstroms
+        pol_embed_border_options.redist_order = core.get_local_option('PE', 'BORDER_REDIST_ORDER')
+        pol_embed_border_options.nredist = core.get_local_option('PE', 'BORDER_N_REDIST')
+        pol_embed_border_options.redist_pol = core.get_local_option('PE', 'BORDER_REDIST_POL')
+
+        pol_embed_options.border_options = pol_embed_border_options
+    return pol_embed_options
+
+
+def psi4mol_to_cppemol(psi4mol):
+    mol = cppe.Molecule()
+    geom = psi4mol.geometry().np
+    for i, c in enumerate(geom):
+        a = cppe.Atom((int(psi4mol.Z(i))), *c)
+        mol.append(a)
+    return mol
+
+
+class CppeInterface:
+    def __init__(self, molecule, options, basisset):
+        # setup the initial CppeState
+        self.molecule = molecule
+        self.options = options
+        self.basisset = basisset
+        self.mints = core.MintsHelper(self.basisset)
+
+        def callback(output):
+            core.print_out("{}\n".format(output))
+
+        self.cppe_state = cppe.CppeState(
+            self.options, psi4mol_to_cppemol(self.molecule), callback
+        )
+        self.cppe_state.calculate_static_energies_and_fields()
+        # obtain coordinates of polarizable sites
+        self._enable_induction = False
+        if self.cppe_state.get_polarizable_site_number():
+            self._enable_induction = True
+            coords = np.array([
+                site.position for site in self.cppe_state.potentials
+                if site.is_polarizable
+            ])
+            self.polarizable_coords = core.Matrix.from_array(coords)
+        self.V_es = None
+
+    def get_pe_contribution(self, density_matrix, elec_only=False):
+        # build electrostatics operator
+        if self.V_es is None and not elec_only:
+            self.build_electrostatics_operator()
+
+        n_bas = self.basisset.nbf()
+        V_pe = np.zeros((n_bas, n_bas))
+        if self._enable_induction:
+            # obtain expectation values of elec. field at polarizable sites
+            elec_fields = self.mints.electric_field_value(
```

Comment:
If you're dedicated to this formatting, I suspect you'll want a `# yapf disable` http://psicode.org/psi4manual/master/prog_style_py.html?highlight=yapf to preserve it. Otherwise, you may want to run `yapf` over the code now to better preserve attribution.

Reply:
I ran `yapf` in the latest commit.